### PR TITLE
include route params in prepare_params result

### PR DESCRIPTION
**Pull request by evgeni**
this is a port of https://github.com/theforeman/foreman-ansible-modules/commit/273e90f34363d67085f97c3e3291d71192a8f120 as I think this ultimately belongs here, and not in FAM.

### CODE REVIEW

**Comment by evgeni** [comment]
the failing FAM tests are due to #78

File: apypie/action.py [hunk 1]
```diff
@@ -227,3 +230,14 @@ def _prepare_params(self, action_params, input_dict):
...
 
         return result
+
+    @staticmethod
+    def _prepare_route_params(route, input_dict):
+        # type: (Route, dict) -> dict
+        result = {}
+
+        for url_param in route.params_in_path:
+            if url_param in input_dict:
```
Comment (mdellweg) [inline apypie/action.py:240]: Should a missing `url_param` not be considered an error?

Comment (evgeni) [reply apypie/action.py:240]: Technically, yes.
But this is getting called with `action.find_best_route(params)`, so this should never happen.

Comment (evgeni) [reply apypie/action.py:240]: But maybe the call to `find_best_route` should be explicit *inside* this function?

Comment (evgeni) [reply apypie/action.py:241]: (updated)

Comment (mdellweg) [reply apypie/action.py:241]: :+1:

So `find_best_route` guarantees that all `params_in_path` are available? Then we really don't need the extra `if`.

Comment (evgeni) [reply apypie/action.py:241]: Yes, it picks the route that consumes as many params as possible.

Comment (evgeni) [reply apypie/action.py]: Turns out, it was needed *scratch head*

Comment (mdellweg) [reply apypie/action.py:241]: ?? So there are optional url_parameter? I'm not going to block on this here, though.

Comment (evgeni) [reply apypie/action.py:241]: FAM tests failed when I dropped the if, and I am not going to find out why now ;)

**Pull request merged**